src/new/wallet.js

```diff
@@ -2469,6 +2470,73 @@ class HathorWallet extends EventEmitter {
 
     return tx;
   }
+
+  /**
+   * Queries the fullnode for a transaction
+   *
+   * @param {string} txId The transaction to query
+   *
+   * @returns {FullNodeTxResponse} Transaction data in the fullnode
+   */
+  async getFullTxById(txId) {
+    const tx = await new Promise(resolve => {
+      txApi.getTransaction(txId, resolve);
+    });
+
+    if (!tx.success) {
+      throw new Error(`Invalid transaction ${txId}`);
+    }
+
+    return tx;
+  }
+
+  /**
+   * Queries the fullnode for a transaction confirmation data
+   *
+   * @param {string} txId The transaction to query
+   *
+   * @returns {FullNodeTxConfirmationDataResponse} Transaction confirmation data
+   */
+  async getTxConfirmationData(txId) {
+    const confirmationData = await new Promise(resolve => {
+      txApi.getConfirmationData(txId, resolve);
+    });
+
+    if (!confirmationData.success) {
+      throw new Error(`Invalid transaction ${txId}`);
+    }
+
+    return confirmationData;
+  }
+
+  /**
+   * Queries the fullnode for a graphviz graph, given a graph type and txId
+   *
+   * @param {string} txId The transaction to query
+   * @param {string} graphType The graph type to query
+   * @param {string} maxLevel Max level to render
+   *
+   * @returns {string} The graphviz digraph
+   */
+  async graphvizNeighborsQuery(
+    txId,
+    graphType,
+    maxLevel,
+  ) {
+    const url = `${config.getServerUrl()}graphviz/neighbours.dot?tx=${txId}&graph_type=${graphType}&max_level=${maxLevel}`;
+    const graphvizData = await new Promise(resolve => {
+      txApi.getGraphviz(url, resolve);
+    });
+
+    // The response will either be a string with the graphviz data or an object
+    // { success: boolean, message: string } so we need to check if the response has
+    // the `success` key
+    if (Object.hasOwnProperty.call(graphvizData, 'success') && !graphvizData.success) {
+      throw new Error(`Invalid transaction ${txId}`);
```

Comment:
From my tests, it will only answer `success: false` and status code 200 when the transaction is invalid (like an arbitrary string), any other invalid param and it will answer with status code 400

Reply:
And how do you handle here a status code 400?

Reply:
Axios will also throw an error, so the handling should be the same.

It will throw the following error:

```
Error: Request failed with status code 400
```